src/auth/discord-auth.service.ts

```diff
@@ -78,4 +80,19 @@ export class DiscordAuthService implements IAuthProvider {
             },
         });
     }
+
+    async findUserByEmails(emails): Promise<any | null> {
```

Comment:
I'm wondering if we can get a better return type for this method? Can we actually return a properly typed user?

If we're going to use `any` as a return type here, you shouldn't need the union with `null`, since any should cover the `null` type.

Reply:
Currently, this is a prisma User Model type, I think there's one we could use, for reference: https://www.prisma.io/docs/orm/prisma-client/type-safety. The current implicit type is this

![image](https://github.com/user-attachments/assets/505f8ef3-ad06-4b70-883f-92d60277017f)